packages/gatsby-plugin-canonical-urls/src/gatsby-ssr.js

```diff
@@ -7,14 +7,19 @@ exports.onRenderBody = (
 ) => {
   if (pluginOptions && pluginOptions.siteUrl) {
     const parsedUrl = url.parse(pluginOptions.siteUrl)
-    const myUrl = `${pluginOptions.siteUrl}${pathname}`
+    const requireTrailingSlash = pluginOptions.requireTrailingSlash || false
+    let myUrl = `${pluginOptions.siteUrl}${pathname}`
+    if(requireTrailingSlash) {
+      myUrl = myUrl.replace(/\/$|$/, `/`)
+    }
     setHeadComponents([
       <link
         rel="canonical"
         key={myUrl}
         href={myUrl}
-        data-baseprotocol={parsedUrl.protocol}
-        data-basehost={parsedUrl.host}
+        data-baseProtocol={parsedUrl.protocol}
+        data-baseHost={parsedUrl.host}
+        data-requireTrailingSlash={`${requireTrailingSlash}`}
```

Comment:
Just thinking out loud here - let me know if this make sense:
Could we perhaps skip adding this attribute if `requireTrailingSlash` isn't true? Not really big deal but would save few bytes for default config and would work the same :)

Reply:
Sounds easy enough, I'll take a stab tonight and update the PR.